generate-assets/src/bin/validate.rs

```diff
@@ -1,45 +1,75 @@
+use std::fmt::Display;
+
+use anyhow::{anyhow, Context, Result};
 use regex::Regex;
 
 use generate_assets::*;
 
-fn main() -> Result<(), ()> {
-    let asset_dir = std::env::args().nth(1).unwrap();
-    if let Ok(asset_root_section) = parse_assets(&asset_dir, None, None, None) {
-        if asset_root_section.validate() {
-            Ok(())
-        } else {
-            Err(())
-        }
-    } else {
-        Err(())
+const MAX_DESCRIPTION_LENGTH: usize = 100;
+
+fn main() -> Result<()> {
+    let asset_dir = std::env::args()
+        .nth(1)
+        .ok_or_else(|| anyhow!("Please specify the path to bevy-assets"))?;
+
+    let asset_root_section =
+        parse_assets(&asset_dir, None, None, None).with_context(|| "Parsing assets")?;
+
+    let results = asset_root_section.validate();
+
+    let errors: Vec<_> = results.iter().filter_map(|r| r.as_ref().err()).collect();
+
+    if errors.is_empty() {
+        return Ok(());
     }
+
+    eprintln!();
+    for error in &errors {
+        eprintln!("{}", error);
+    }
+
+    Err(anyhow!("{} asset(s) are invalid.", errors.len()))
```

Comment:
you can use the bail! macro from anyhow for this
```suggestion
    bail!("{} asset(s) are invalid.", errors.len())
```

Reply:
It's an anyhow macro, so clippy wouldn't know about it. I'm honestly not sure how idiomatic that is. I just saw the Err(anyhow and thought of the macro. I don't mind either way honestly it was just going through my head,